cmd/syncComparator.go

```diff
@@ -57,11 +63,23 @@ func (f *syncDestinationComparator) processIfNecessary(destinationObject StoredO
 	// if the destinationObject is present at source and stale, we transfer the up-to-date version from source
 	if present {
 		defer delete(f.sourceIndex.indexMap, destinationObject.relativePath)
-		if f.disableComparison || sourceObjectInMap.isMoreRecentThan(destinationObject) {
-			err := f.copyTransferScheduler(sourceObjectInMap)
-			if err != nil {
-				return err
+
+		if f.disableComparison {
+			return f.copyTransferScheduler(sourceObjectInMap)
+		}
+
+		if f.comparisonHashType != common.ESyncHashType.None() {
+			switch f.comparisonHashType {
+			case common.ESyncHashType.MD5():
+				if !reflect.DeepEqual(sourceObjectInMap.md5, destinationObject.md5) {
+					// hash inequality = source "newer" in this model.
+					return f.copyTransferScheduler(sourceObjectInMap)
+				}
```

Comment:
For hash type match also we can dump a debug log that we are skipping this file. In future if someone reports and issue with the sync it will be easy to identify what we skipped and what we synced.

Reply:
Sure. I'll add some debug logging in.